### DISCUSSION TITLE
Running time: scale down input or prune with constraint?

### DISCUSSION THREAD

**Discussion by rybickibartek** [google/or-tools#3229] [CP-SAT questions] [answered]
Hi,

i have some number of int variables and each of them is coupled with constraint AddModuloEquality(0, var, 5).

I can adjust input which will allow me to define int variables with five times smaller domain and, in the consequence, remove AddModuloEquality constraint.

Question is the following: does it make sense? I am thinking about running time, because obtained result will be the same.

Bartek

**Comment by lperron** [comment, marked as answer]
yes. or just create the domains with only multiple of 5. The solver will do the rescaling.

**Comment by rybickibartek** [comment]
Thanks for answer, so to speed up computations i will create the domains with only multiple of 5 as you suggest.
